File: backend/Modules/Cursos/CursosExceptions.cs
namespace CodeMX.Api.Modules.Cursos;

/// <summary>Se lanza cuando el alumno intenta el examen sin el avance mínimo. Mapea a HTTP 403.</summary>
public class ExamenBloqueadoException : Exception
{
    public ExamenBloqueadoException(string message) : base(message) { }
}

/// <summary>Recurso de cursos no encontrado. Mapea a HTTP 404.</summary>
public class RecursoNoEncontradoException : Exception
{
    public RecursoNoEncontradoException(string message) : base(message) { }
}

File: backend-java/src/main/java/mx/codemx/modules/cursos/CursosExceptions.java
package mx.codemx.modules.cursos;

public final class CursosExceptions {

    private CursosExceptions() {
    }

    public static class ExamenBloqueadoException extends RuntimeException {
        public ExamenBloqueadoException(String mensaje) {
            super(mensaje);
        }
    }

    public static class RecursoNoEncontradoException extends RuntimeException {
        public RecursoNoEncontradoException(String mensaje) {
            super(mensaje);
        }
    }
}
